resim/metrics/python/metrics_writer.py

```diff
@@ -57,6 +60,16 @@ def add_metric(self, metric: MetricT) -> MetricT:
         self.metrics[metric.id] = metric
         return metric
 
+    def base_add_event(self, event: Event) -> Event:
+        assert event.name not in self.names
+        self.names.add(event.name)
+        self.events[event.id] = event
+        return event
+
+    def add_event(self, name: str) -> Event:
+        event = Event(name=name)
+        return self.base_add_event(event)
```

Comment:
[nit] Would you mind adding some docstrings here? I think we should have docstrings on a lot more of these methods, but in this case, it was initially confusing to me when you might use the base_ variant vs. the standard one.

Reply:
yeah, good point